### ISSUE TITLE
Content produced diagnostics when parsed

### ISSUE THREAD

**Issue by Matkurban** [Workiva/dependency_validator#189]
admin@admindeMacBook-Pro Elephant % flutter analyze
Analyzing Elephant...                                                   
No issues found! (ran in 6.0s)
admin@admindeMacBook-Pro Elephant % dart pub global run dependency_validator

Validating dependencies for elephant...
Error parsing: ./lib/utils/file_util.dart
Content produced diagnostics when parsed:
  extraneous_modifier: Can't have modifier 'final' here. - 79:31

**Comment by matthewnitschke-wk** [comment]
We'll need some more context before we can help

Can you share the content of `file_util.dart`? what language of dart is being used (`dart --version`)? what version of dependency_validator is installed (`dart pub global list`)?

**Comment by Matkurban** [comment]
[✓] Flutter (Channel stable, 3.44.9, on macOS 26.6.2 25G83 darwin-arm64, locale zh-Hans-CN) [366ms]
    • Flutter version 3.44.9 on channel stable at /Users/admin/develop/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 6b182d2c75 (5 weeks ago), 2026-08-05 10:04:07 -0700
    • Engine revision 5a2a6a42cc
    • Dart version 3.12.2
    • DevTools version 2.57.0
    • Feature flags: enable-web, enable-linux-desktop, enable-macos-desktop, enable-windows-desktop, enable-android, enable-ios, cli-animations, enable-native-assets, enable-swift-package-manager,
      omit-legacy-version-file, enable-lldb-debugging, enable-uiscene-migration

[!] Android toolchain - develop for Android devices (Android SDK version 37.0.0) [467ms]
    • Android SDK at /Users/admin/Library/Android/sdk
    • Emulator version 37.1.11.0 (build_id 15917651) (CL:N/A)
    • Platform android-37.1, build-tools 37.0.0
    • Java binary at: /Users/admin/develop/java/jdk-17.0.20.jdk/Contents/Home/bin/java
      This JDK is specified in your Flutter configuration.
      To change the current JDK, run: `flutter config --jdk-dir="path/to/jdk"`.
    • Java version Java(TM) SE Runtime Environment (build 17.0.20+7-LTS-191)
    ✗ Android license status unknown.
      Run `flutter doctor --android-licenses` to accept the SDK licenses.
      See https://flutter.dev/to/macos-android-setup for more details.

[✓] Xcode - develop for iOS and macOS (Xcode 26.6) [717ms]
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 17F113
    • CocoaPods version 1.17.0

[✓] Chrome - develop for the web [8ms]
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Connected device (3 available) [6.0s]
    • iPhone (mobile) • 00008030-000511383498802E • ios            • iOS 26.6.1 23G83
    • macOS (desktop) • macos                     • darwin-arm64   • macOS 26.6.2 25G83 darwin-arm64
    • Chrome (web)    • chrome                    • web-javascript • Google Chrome 153.0.8010.36

[✓] Network resources [1,324ms]
    • All expected network resources are available.

! Doctor found issues in 1 category.

**Comment by Matkurban** [comment]
```dart
import 'package:background_downloader/background_downloader.dart';
import 'package:flutter_toast_pro/flutter_toast_pro.dart';
import 'package:mime/mime.dart';
import 'package:elephant/config/app_config.dart';
import 'package:elephant/constant/constants.dart';
import 'package:elephant/constants/language.dart';
import 'package:elephant/utils/permission_util.dart';
import 'package:path/path.dart' as p;
import 'package:xue_hua_file_operations/xue_hua_file_operations.dart';
import 'package:xue_hua_sdk/xue_hua_sdk.dart';

enum CustomFileType { none, image, video }

sealed class FileUtil {
  ///获取文件类型
  static CustomFileType getFileType(String path) {
    final String? mimeType = lookupMimeType(path);
    if (mimeType != null) {
      final CustomFileType? fromMime = _fileTypeFromExtension(extensionFromMime(mimeType));
      if (fromMime != null) {
        return fromMime;
      }
    }
    return _fileTypeFromExtension(p.extension(path).replaceFirst('.', '')) ?? CustomFileType.none;
  }

  static CustomFileType? _fileTypeFromExtension(String? ext) {
    if (ext == null || ext.isEmpty) return null;
    switch (ext.toLowerCase()) {
      case 'png':
      case 'jpg':
      case 'jpeg':
      case 'jpe':
      case 'jpgv':
      case 'gif':
      case 'webp':
        return CustomFileType.image;
      case 'mp4':
      case 'mpg':
      case 'mpeg':
      case 'mov':
      case 'mkv':
      case 'avi':
        return CustomFileType.video;
      default:
        return null;
    }
  }

  static CustomFileType getFileTypeFromString(String? mimeType) {
    if (mimeType != null) {
      var fromMime = extensionFromMime(mimeType);
      if (fromMime != null) {
        if (fromMime == "png" ||
            fromMime == "jpg" ||
            fromMime == "jpeg" ||
            fromMime == "jpe" ||
            fromMime == "jpgv") {
          return CustomFileType.image;
        } else if (fromMime == "mp4" ||
            fromMime == "m4a" ||
            fromMime == "m4b" ||
            fromMime == "mp4a" ||
            fromMime == "mpg" ||
            fromMime == "mov" ||
            fromMime == "mkv") {
          return CustomFileType.video;
        } else {
          return CustomFileType.none;
        }
      } else {
        return CustomFileType.none;
      }
    } else {
      return CustomFileType.none;
    }
  }

  static String? getMediaType(final String filePath) {
    var fileName = filePath.substring(filePath.lastIndexOf("/") + 1);
    var fileExt = fileName.substring(fileName.lastIndexOf("."));
    switch (fileExt.toLowerCase()) {
      case ".jpg":
      case ".jpeg":
      case ".jpe":
        return "image/jpeg";
      case ".png":
        return "image/png";
      case ".bmp":
        return "image/bmp";
      case ".gif":
        return "image/gif";
      case ".json":
        return "application/json";
      case ".svg":
      case ".svgz":
        return "image/svg+xml";
      case ".mp3":
        return "audio/mpeg";
      case ".mp4":
        return "video/mp4";
      case ".mov":
        return "video/mov";
      case ".htm":
      case ".html":
        return "text/html";
      case ".css":
        return "text/css";
      case ".csv":
        return "text/csv";
      case ".txt":
      case ".text":
      case ".conf":
      case ".def":
      case ".log":
      case ".in":
        return "text/plain";
    }
    return null;
  }

  ///保存网络图片
  static Future<void> saveNetworkImage(String url, {String? fileName}) async {
    GalleryPermissionStatus status = await XueHuaFileOperations.instance.galleryPermissionStatus(
      forAlbum: true,
    );
    if (status.isPermanentlyDenied || status.isRestricted) {
      final granted = await XueHuaFileOperations.instance.requestGalleryPermission();
      if (granted.isPermanentlyDenied || granted.isRestricted) {
        FlutterToastPro.error(l10n.saveFailed);
        return;
      }
    }
    final String filename = '${fileName ?? DateTime.now().millisecondsSinceEpoch}.png';
    final DownloadTask task = DownloadTask(
      url: url,
      filename: filename,
      directory: 'image',
      baseDirectory: BaseDirectory.temporary,
      updates: Updates.statusAndProgress,
      retries: 5,
      allowPause: true,
    );
    try {
      final TaskStatusUpdate result = await FileDownloader().download(
        task,
        onProgress: (double progress) {
          if (progress < 0) return;
          FlutterToastPro.progress(progress);
        },
      );
      if (result.status != TaskStatus.complete) {
        error('保存网络图片失败: ${result.status} ${result.exception?.description ?? ''}');
        FlutterToastPro.error(l10n.downloadFailed);
        return;
      }
      final String imagePath = await task.filePath();
      await XueHuaFileOperations.instance.saveToGallery(
        fileName: filename,
        sourcePath: imagePath,
        albumName: AppConfig.albumName,
      );
      FlutterToastPro.success(appLocalizations.saveSuccess);
    } catch (e, s) {
      error('保存网络图片失败', error: e, stackTrace: s);
      FlutterToastPro.error(l10n.downloadFailed);
    } finally {
      FlutterToastPro.hideProgress();
    }
  }

  ///保存图片消息
  static Future<void> saveImageMessage(Message message) async {
    if (!await PermissionUtil.albumPermission()) {
      FlutterToastPro.error(appLocalizations.saveFailed);
      return;
    }

    String? sourceUrl = message.pictureElem?.sourcePicture?.url;
    if (sourceUrl != null && sourceUrl.isNotEmpty) {
      await saveNetworkImage(sourceUrl);
    } else {
      FlutterToastPro.error(appLocalizations.imageExpiredOrInvalid);
    }
  }

  /// 保存网络视频到相册
  static Future<void> saveNetworkVideo(String url, {String? fileName}) async {
    GalleryPermissionStatus status = await XueHuaFileOperations.instance.galleryPermissionStatus(
      forAlbum: true,
    );

    if (status.isPermanentlyDenied || status.isRestricted) {
      final granted = await XueHuaFileOperations.instance.requestGalleryPermission();
      if (granted.isPermanentlyDenied || granted.isRestricted) {
        FlutterToastPro.error(l10n.saveFailed);
        return;
      }
    }
    final String filename = '${fileName ?? DateTime.now().millisecondsSinceEpoch}.mp4';
    final DownloadTask task = DownloadTask(
      url: url,
      filename: filename,
      directory: 'video',
      baseDirectory: BaseDirectory.temporary,
      updates: Updates.statusAndProgress,
      retries: 5,
      allowPause: true,
    );
    try {
      final TaskStatusUpdate result = await FileDownloader().download(
        task,
        onProgress: (double progress) {
          if (progress < 0) return;
          FlutterToastPro.progress(progress);
        },
      );
      if (result.status != TaskStatus.complete) {
        error('保存网络视频失败: ${result.status} ${result.exception?.description ?? ''}');
        FlutterToastPro.error(l10n.downloadFailed);
        return;
      }
      final String videoPath = await task.filePath();
      await XueHuaFileOperations.instance.saveToGallery(
        fileName: filename,
        sourcePath: videoPath,
        albumName: AppConfig.albumName,
      );
      FlutterToastPro.success(appLocalizations.saveSuccess);
    } catch (e, s) {
      error('保存网络视频失败', error: e, stackTrace: s);
      FlutterToastPro.error(l10n.downloadFailed);
    } finally {
      FlutterToastPro.hideProgress();
    }
  }

  /// 保存视频消息到相册
  static Future<void> saveVideoMessage(Message message) async {
    if (!await PermissionUtil.albumPermission()) {
      FlutterToastPro.error(appLocalizations.saveFailed);
      return;
    }
    String? videoUrl = message.videoElem?.videoUrl;
    if (videoUrl != null && videoUrl.isNotEmpty) {
      await saveNetworkVideo(videoUrl);
    } else {
      FlutterToastPro.error(appLocalizations.videoExpiredOrInvalid);
    }
  }

  static bool isNetworkAddress(String path) {
    final lowerCasePath = path.toLowerCase();
    return lowerCasePath.startsWith('http://') || lowerCasePath.startsWith('https://');
  }
}
```

**Comment by matthewnitschke-wk** [comment]
in your example I see

```dart
static String? getMediaType(final String filePath) {
```

This should be resolved by: https://github.com/Workiva/dependency_validator/pull/187
